File: source/frontend/src/modules/tasks/types/task.request.ts
import { TaskPriority, TaskStatus } from './task.enum'

export interface CreateTaskRequest {
  title: string
  description?: string
  priority?: TaskPriority
  startDate?: string
  dueDate?: string | null
  assignedTo?: number | null
  labelIds?: number[]
}

export interface UpdateTaskRequest {
  title?: string
  description?: string
  priority?: TaskPriority
  startDate?: string
  dueDate?: string | null
  assignedTo?: number | null
}

export interface UpdateTaskStatusRequest {
  newStatus: TaskStatus
  note?: string
}

export interface AssignTaskRequest {
  assignedTo: number | null
}

export interface AddTaskLabelRequest {
  labelId: number
}

export interface RemoveTaskLabelRequest {
  labelId: number
}

File: source/backend/project.Presentation/Models/TaskRequest.cs
using project.Domain.Models;

namespace project.Presentation.Models
{
    public class CreateTaskRequest
    {
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public TaskPriority Priority { get; set; } = TaskPriority.Medium;
        public TasksStatus TaskStatus { get; set; }
        public int? AssignedTo { get; set; }
        public DateTime? DueDate { get; set; }
    }

    public class UpdateTaskRequest
    {
        public string Title { get; set; }
        public string? Description { get; set; }
        public TaskPriority Priority { get; set; }
        public TasksStatus TaskStatus { get; set; }
        public int? AssignedTo { get; set; }
        public DateTime? DueDate { get; set; }
    }

    public class UpdateTaskStatusRequest
    {
        public TasksStatus NewStatus { get; set; }
    }

    public class AssignTaskRequest
    {
        public int AssignedTo { get; set; } // null = unassign
    }

    public class AddTaskLabelRequest
    {
        public int LabelId { get; set; }
    }

    public class RemoveTaskLabelRequest
    {
        public int LabelId { get; set; }
    }
    public class SetDueDateRequest
    {
        public DateTime DateTime { get; set; }
    }
}
